scripts/distributed_pm.py

```diff
@@ -0,0 +1,105 @@
+import os
+
+os.environ["EQX_ON_ERROR"] = "nan"  # avoid an allgather caused by diffrax
+import jax
+
+jax.distributed.initialize()
+
+rank = jax.process_index()
+size = jax.process_count()
+
+import jax.numpy as jnp
+import jax_cosmo as jc
+import numpy as np
+from diffrax import Dopri5, ODETerm, PIDController, SaveAt, diffeqsolve
+from jax.experimental import mesh_utils
+from jax.sharding import Mesh, NamedSharding
+from jax.sharding import PartitionSpec as P
+
+from jaxpm.kernels import interpolate_power_spectrum
+from jaxpm.painting import cic_paint_dx
+from jaxpm.pm import linear_field, lpt, make_ode_fn
+
+size = 256
+mesh_shape = [size] * 3
+box_size = [float(size)] * 3
+snapshots = jnp.linspace(0.1, 1., 4)
+halo_size = 64
+if jax.device_count() > 1:
+
+    pdims = (4, 2)
+    devices = mesh_utils.create_device_mesh(pdims)
+    mesh = Mesh(devices.T, axis_names=('x', 'y'))
+    sharding = NamedSharding(mesh, P('x', 'y'))
+
+
+@jax.jit
+def run_simulation(omega_c, sigma8):
+    # Create a small function to generate the matter power spectrum
+    k = jnp.logspace(-4, 1, 128)
+    pk = jc.power.linear_matter_power(
+        jc.Planck15(Omega_c=omega_c, sigma8=sigma8), k)
+
+    pk_fn = lambda x: interpolate_power_spectrum(x, k, pk)
+    # Create initial conditions
+    initial_conditions = linear_field(mesh_shape,
+                                      box_size,
+                                      pk_fn,
+                                      seed=jax.random.PRNGKey(0))
+
+    # Create particles
+    particles = jnp.stack(jnp.meshgrid(*[jnp.arange(s) for s in mesh_shape]),
+                          axis=-1).reshape([-1, 3])
```

Comment:
```suggestion
```

Reply:
about that
Initial particles are used both in LPT and LPT2
They are created as a distributed (X, Y , Z , 3)
Does not make sense to reuse them?
Also some users might not want to start from uniform particle positions

Reply:
yeah but 
1. it's not used in this script
2. it's not distributed here, so crashes if too large